mailpoet/lib/Cron/Workers/Scheduler.php

```diff
@@ -129,23 +150,25 @@ public function processPostNotificationNewsletter($newsletter, SendingTask $queu
 
     $newsletterEntity = $this->newslettersRepository->findOneById($newsletter->id);
 
-    if ($newsletterEntity instanceof NewsletterEntity) {
-      // ensure that segments exist
-      $segments = $newsletterEntity->getSegmentIds();
-      if (empty($segments)) {
-        $this->loggerFactory->getLogger(LoggerFactory::TOPIC_POST_NOTIFICATIONS)->info(
-          'post notification no segments',
-          ['newsletter_id' => $newsletter->id, 'task_id' => $queue->taskId]
-        );
-        return $this->deleteQueueOrUpdateNextRunDate($queue, $newsletter);
-      }
+    if (!$newsletterEntity instanceof NewsletterEntity) {
+      throw new InvalidStateException();
```

Comment:
I don't believe the previous version of this code ever threw any exceptions. Could this cause an issue since this method is getting called in a `foreach` that's potentially processing multiple queues? In other words, could it throw an unhandled exception that would prevent the next newsletters from getting processed?

Reply:
This is a good point, but the exception shouldn't be thrown at any time. We know a newsletter exists in the DB, but we need to load it for Doctrine. Because we refactor the code by parts, we sometimes need to have objects from the old and the new model.
When the whole class is refactored to Doctrine, this check can be removed because the new model will be passed from an argument.

If you still think this solution is too dangerous, I can wrap creating post notifications with `if ($newsletterEntity)` and remove this exception.

Reply:
That makes sense, and thanks for the explanation!

Knowing that we'll be able to remove this once the whole class gets refactored makes me feel much better.